src/Maestro/Maestro.MergePolicies/AllChecksSuccessfulMergePolicy.cs

```diff
@@ -26,6 +26,8 @@ public AllChecksSuccessfulMergePolicy(HashSet<string> ignoreChecks)
 
     public override string DisplayName => "All Checks Successful";
 
+    public override string Name => "AllChecksSuccessful";
```

Comment:
I don't follow what is the actual change. Aren't these going to evaluate to the same string as before?

Reply:
Basically I am asking how will this fix anything since it seems like all the `Name` properties will have the same value as before.

Reply:
The fix works because of the way it was implemented before. We were just stripping the `MergePolicy` from the class name. Because of this the following happened:
- We would go into the CodeFlowPolicy, check if there are any `CodeFlow` cached results, there are non because it's the first time we're calling this
- It would call the Forward flow merge policy to evaluate
- it would return a result with the `ForwardFlow` name
- Next time, we'd go into the MergePolicyEvaluator, check if there's a policy result called `CodeFlow`, but there's not, since we have `ForwardFlow` in there instead
- We'd add another `ForwardFlow`
- Then the next time, we'd get the error when constructing the Dictionary

Reply:
Can't we keep the original `Name` implementation, make it `virtual` and only overwrite it in the code flow policies?

Reply:
that would work too, but my thoughts were that this would be more explicit. The way it is now, it's a bit hidden, and could be unpredictable.